Content.Server/Resist/EscapeInventorySystem.cs

```diff
@@ -26,20 +34,32 @@ public override void Initialize()
         SubscribeLocalEvent<CanEscapeInventoryComponent, UpdateCanMoveEvent>(OnMoveAttempt);
         SubscribeLocalEvent<CanEscapeInventoryComponent, EscapeDoAfterComplete>(OnEscapeComplete);
         SubscribeLocalEvent<CanEscapeInventoryComponent, EscapeDoAfterCancel>(OnEscapeFail);
+        SubscribeLocalEvent<CanEscapeInventoryComponent, DroppedEvent>(OnDropped);
     }
 
     private void OnRelayMovement(EntityUid uid, CanEscapeInventoryComponent component, ref MoveInputEvent args)
     {
-        //Prevents the user from creating multiple DoAfters if they're already resisting.
-        if (component.IsResisting == true)
+        if (component.CancelToken != null)
+            return;
+
+        if (!_containerSystem.TryGetContainingContainer(uid, out var container) || !_actionBlockerSystem.CanInteract(uid, container.Owner))
             return;
 
-        if (_containerSystem.TryGetContainingContainer(uid, out var container)
-            && (HasComp<SharedStorageComponent>(container.Owner) || HasComp<InventoryComponent>(container.Owner) || HasComp<SharedHandsComponent>(container.Owner)))
+        // Contested
+        if (_handsSystem.IsHolding(container.Owner, uid, out var inHand))
         {
-            if (_actionBlockerSystem.CanInteract(uid, container.Owner))
-                AttemptEscape(uid, container.Owner, component);
+            var contestResults = 1 / (_contests.MassContest(uid, container.Owner));
```

Comment:
Contests code needs fixing to not throw errors if it doesn't resolve.

This also doesn't account for 0 mass on the uid.